tests-ui/tests/store/modelToNodeStore.test.ts

```diff
@@ -0,0 +1,245 @@
+  // the only property ModelNodeProvider constructor uses and tests verify
+  const createMockNodeDef = (name: string): ComfyNodeDefImpl => {
+    return { name } as ComfyNodeDefImpl
+  }
+
+  beforeEach(() => {
+    setActivePinia(createPinia())
+    store = useModelToNodeStore()
+    nodeDefStore = useNodeDefStore()
+
+    const mockNodeNames = [
+      'CheckpointLoaderSimple',
+      'ImageOnlyCheckpointLoader',
+      'LoraLoader',
+      'LoraLoaderModelOnly',
+      'VAELoader',
+      'ControlNetLoader',
+      'UNETLoader',
+      'UpscaleModelLoader',
+      'StyleModelLoader',
+      'GLIGENLoader'
+    ]
+
+    const mockNodeDefs: Record<string, ComfyNodeDefImpl> = Object.fromEntries(
+      mockNodeNames.map((name) => [name, createMockNodeDef(name)])
+    )
+
+    nodeDefStore.nodeDefsByName = mockNodeDefs
+  })
+
+  describe('modelToNodeMap', () => {
+    it('should initialize as empty', () => {
+      expect(Object.keys(store.modelToNodeMap)).toHaveLength(0)
+    })
+
+    it('should populate after registration', () => {
+      store.registerDefaults()
+      expect(Object.keys(store.modelToNodeMap)).toContain('checkpoints')
+      expect(Object.keys(store.modelToNodeMap)).toContain('unet')
+    })
+  })
+
+  describe('getNodeProvider', () => {
+    it('should return provider for registered model type', () => {
+      store.registerDefaults()
+
+      const provider = store.getNodeProvider('checkpoints')
+      expect(provider).toBeDefined()
+      // Optional chaining used because getNodeProvider() can return undefined for unregistered types
+      expect(provider?.nodeDef.name).toBe('CheckpointLoaderSimple')
+      expect(provider?.key).toBe('ckpt_name')
+    })
+
+    it('should return undefined for unregistered model type', () => {
+      store.registerDefaults()
+      expect(store.getNodeProvider('nonexistent')).toBeUndefined()
+    })
+
+    it('should return first registered provider when multiple providers exist for same model type', () => {
+      store.registerDefaults()
+
+      const provider = store.getNodeProvider('checkpoints')
+      // Using optional chaining for safety since getNodeProvider() can return undefined
+      expect(provider?.nodeDef.name).toBe('CheckpointLoaderSimple')
+    })
+
+    it('should trigger lazy registration when called before registerDefaults', () => {
+      const provider = store.getNodeProvider('checkpoints')
+      expect(provider).toBeDefined()
+    })
+  })
+
+  describe('getAllNodeProviders', () => {
+    it('should return all providers for model type with multiple nodes', () => {
+      store.registerDefaults()
+
+      const checkpointProviders = store.getAllNodeProviders('checkpoints')
+      expect(checkpointProviders).toHaveLength(2)
+      expect(checkpointProviders.map((p) => p.nodeDef.name)).toContain(
+        'CheckpointLoaderSimple'
+      )
+      expect(checkpointProviders.map((p) => p.nodeDef.name)).toContain(
+        'ImageOnlyCheckpointLoader'
+      )
+
```

Comment:
For all of this style assertion: https://vitest.dev/api/expect.html#expect-objectcontaining